plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelsAction.java

```diff
@@ -156,11 +163,48 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLUnde
     private void undeployModels(String[] targetNodeIds, String[] modelIds, ActionListener<MLUndeployModelsResponse> listener) {
         MLUndeployModelNodesRequest mlUndeployModelNodesRequest = new MLUndeployModelNodesRequest(targetNodeIds, modelIds);
 
-        client.execute(MLUndeployModelAction.INSTANCE, mlUndeployModelNodesRequest, ActionListener.wrap(r -> {
-            listener.onResponse(new MLUndeployModelsResponse(r));
+        client.execute(MLUndeployModelAction.INSTANCE, mlUndeployModelNodesRequest, ActionListener.wrap(response -> {
+            if (response.getNodes().isEmpty()) {
```

Comment:
[nit]Let's add a detailed explanation here to ensure that future developers understand the reason behind this logic and aren't confused about why it's implemented this way.